Add teamd patches to solve traffic loss issue when removing port from LAG

liorghub opened the pull request:
<!--
     Please make sure you've read and understood our contributing guidelines:
     https://github.com/Azure/SONiC/blob/gh-pages/CONTRIBUTING.md

     ** Make sure all your commits include a signature generated with `git commit -s` **

     If this is a bug fix, make sure your description includes "fixes #xxxx", or
     "closes #xxxx" or "resolves #xxxx"

     Please provide the following information:
-->

#### Why I did it
When removing port from LAG while traffic is running thorough LAG there is traffic disruption of 60 seconds.

#### How I did it
The patch I added introduces "port_removing" op and call it right before Kernel is asked to remove the port. 
Implement the op in LACP runner to disable the port which leads to proper LACPDU send.

#### How to verify it
Set LAG between 2 switches.
Set LAGs to be router port and set ip address.
In switch A send ping to ip address of LAG in switch B.
In switch B, while ping is running remove port from LAG.
Verify ping is not stopping.

#### Which release branch to backport (provide reason below if selected)

<!--
- Note we only backport fixes to a release branch, *not* features!
- Please also provide a reason for the backporting below.
- e.g.
- [x] 202006
-->

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [x] 202205
- [x] 202211

#### Description for the changelog
<!--
Write a short (one line) summary that describes the changes in this
pull request for inclusion in the changelog:
-->

#### Ensure to add label/tag for the feature raised. example - PR#2174 under sonic-utilities repo. where, Generic Config and Update feature has been labelled as GCU.

#### Link to config_db schema for YANG module changes
<!--
Provide a link to config_db schema for the table for which YANG model
is defined
Link should point to correct section on https://github.com/Azure/sonic-buildimage/blob/master/src/sonic-yang-models/doc/Configuration.md
-->

#### A picture of a cute animal (not mandatory but encouraged)

liat-grozovik commented:
@stepanblyschak , @vaibhavhd  appreciate if you can review.

liat-grozovik commented:
@liorghub can you confirm this can be cleanly cherry picked to 202205?

liorghub commented:
@liat-grozovik 
> @liorghub can you confirm this can be cleanly cherry picked to 202205?

Yes, this PR can be cleanly cherry picked to 202205.

liorghub commented:
This PR fixes issue https://github.com/sonic-net/sonic-buildimage/issues/14381

saiarcot895 commented on src/libteam/patch/0014-dont-move-the-port-state-from-disabled-when-admin-state-is-down:
Can you add a `.patch` suffix to this?

saiarcot895 commented on src/libteam/patch/series:
```diff
@@ -10,3 +10,5 @@
 0011-Remove-extensive-debug-output.patch
 0012-Increase-min_ports-upper-limit-to-1024.patch
+0013-set-port-to-disabled-state-during-removal.patch
+0014-dont-move-the-port-state-from-disabled-when-admin-state-is-down
```
This needs to be updated.

liorghub replied:
Apologies, done.

saiarcot895 commented:
I tried to repro the issue described here on a t0-64-32 topo with sonic neighbors (with config_db.json modified to remove the `"min_links": 2` requirement), and didn't see any ping failures while I removed and added a LAG member from an existing LAG. I did see slightly higher RTT times (probably because the primary interface was being removed, and another interface had to be selected).

Is this happening only on physical hardware, or is it on a virtual setup as well?

liorghub commented:
> I tried to repro the issue described here on a t0-64-32 topo with sonic neighbors (with config_db.json modified to remove the `"min_links": 2` requirement), and didn't see any ping failures while I removed and added a LAG member from an existing LAG. I did see slightly higher RTT times (probably because the primary interface was being removed, and another interface had to be selected).
> 
> Is this happening only on physical hardware, or is it on a virtual setup as well?

It happens only on physical hardware when both sides are switches (when one side is host and we create bonding, issue is not reproduced).

saiarcot895 reviewed (approved):
I verified that this is sort-of happening on virtual switch as well. The ICMP response packet can arrive on the interface that was just removed from the port channel (and so outside of the port channel), if it happens to be the primary interface on the peer virtual switch. It doesn't manifest as a traffic drop because `rp_filter` checks are disabled, so the kernel accepts it as a valid packet, even though it's outside of the port channel.

mssonicbld commented:
Cherry-pick PR to 202211: https://github.com/sonic-net/sonic-buildimage/pull/14684

mssonicbld commented:
Cherry-pick PR to 202205: https://github.com/sonic-net/sonic-buildimage/pull/14716
